File: src/main/java/br/com/objective/gourmetgame/Response.java
package br.com.objective.gourmetgame;

import javax.swing.*;

public class Response implements Question {

    public static final String TEXT_QUESTION = "O prato que você pensou é %s?";
    private String information;
    private Question responsePositive = new EmptyResponse();
    private Question responseNegative = new EmptyResponse();

    public Response(String information){
        this.information = information;
    }

    public void changePositiveAndNegative(String positive, String negative){
        this.responsePositive = new Response(positive);
        this.responseNegative = new Response(negative);
    }

    @Override
    public void quiz() {
        String textQuestion = String.format(TEXT_QUESTION, this.information);
        int response = JOptionPane.showConfirmDialog(null, textQuestion , titleQuestion, JOptionPane.YES_NO_OPTION);
        switch (response){
            case JOptionPane.YES_OPTION:
                responsePositive.positive();
                break;
            default:
                responseNegative.negative(this);
                break;
        }
    }

    @Override
    public void positive() {
        this.quiz();
    }

    @Override
    public void negative(Question question) {
        this.quiz();
    }

    @Override
    public void addQuestion(String otherDish,String difference) {
        responseNegative = new Response(this.information);
        responsePositive = new Response(otherDish);
        this.information = difference;
    }

    @Override
    public String getInformation() {
        return information;
    }

}

File: src/app/model/response.ts
import { Question } from "src/app/model/question";
import { EmptyResponse } from "src/app/model/emptyResponse";
import { MatDialog } from "@angular/material";
import { DialogComfirmComponent } from "src/app/dialog-comfirm/dialog-comfirm.component";

export class Response implements Question {

    private information:string;
    private responsePositive:Question = new EmptyResponse(this.dialog);
    private responseNegative:Question = new EmptyResponse(this.dialog);

    constructor(information: string,private dialog:MatDialog) {
        this.information = information;
    }

    public changePositiveAndNegative(positive:string, negative:string) : void {
        this.responsePositive = new Response(positive, this.dialog);
        this.responseNegative = new Response(negative, this.dialog);
    }
    
    quiz() {
         this.dialog.open(DialogComfirmComponent, {
            data: {
                message: `O prato que você pensou é ${this.information}?`
            }, disableClose : true
        }).afterClosed().subscribe(data=>{
            data ? this.responsePositive.positive() : this.responseNegative.negative(this);
        });
    }
    positive() {
        this.quiz();
    }
    negative(question: Question) {
        this.quiz();
    }
    addQuestion(otherDish: string, difference: string) {
        this.responseNegative = new Response(this.information, this.dialog);
        this.responsePositive = new Response(otherDish, this.dialog);
        this.information = difference;
    }

    getInformation(): string {
        return this.information;
    }

}